src/Exception.php

```diff
@@ -65,6 +65,10 @@ public static function createFromCode($code, $messagePrefix = 'Socket error')
      */
     protected static function getErrorMessage($code)
     {
+        if (null === $code || false === $code) {
+            return 'not a valid resource';
+        }
```

Comment:
IMO this check already violates the method argument definition: This method only accepts an `int`, so I think this logic should be moved somewhere else.

Also not a fan of hard-coding these strings, how about using `SOCKET_ENOTSOCK` (or something else from https://www.php.net/manual/en/sockets.constants.php)?